_includes/datasets_display.html

```diff
@@ -0,0 +1,72 @@
+
+{% comment %}Only display datasets in current category if there is a current category{% endcomment %}
+{% if current_category %}
+    {% assign current_slug = current_category | slugify %}
+    {% assign datasets = "" | split: "" %}
+    {% for s in site.datasets %}
+        {% capture slugs %}{% for category in s.categories %}{{category|slugify}}|{%endfor%}{%endcapture%}
+        {% assign site_slugs = slugs | split: "|" %}
+        {% if site_slugs contains current_slug %}
+            {% assign datasets = datasets | push: s %}
+        {% endif %}
+    {% endfor %}
+{% else %}
+    {% assign datasets = site.datasets %}
+{% endif %}
+
+{% assign datasets_count = datasets | size %}
+
+<div class="row">
+  <div class="col-sm-4">
+
+    <h3>Categories</h3>
+
+    {% assign featured_categories = site.data.categories | where: 'featured', 'true' %}
+    {% capture featured_names %}{% for c in featured_categories%}{{c.name}},{% endfor %}{% endcapture %}
+    {% assign highlighted = featured_names | split: "," %}
+    {% if current_category %}
+        {% unless highlighted contains current_category %}
+            {% assign highlighted = highlighted | push: current_category %}
+        {% endunless %}
+    {% endif %}
+    <div class="list-group" data-component="categories-filter" data-show="5">
+
+      {% include filter-category.html categories=highlighted current_category=current_category %}
+
+      {% assign categories_all = site.datasets | map: 'category' | uniq %}
+      {% assign categories = '' | split: "" %}
+      {% for category in categories_all %}
+        {% unless highlighted contains category %}
+          {% assign categories = categories | push: category %}
+        {% endunless %}
+      {% endfor %}
+      <div class="js-hidden">
+      {% include filter-category.html categories=categories current=current_category  %}
```

Comment:
This says `current` instead of `current_categoy`, but works because you don't need to mention variables if they're the same name, context is passed down (in fact, variables included here are found under `include.` so I don't think work as you think they do).

Reply:
I think I'm thinking of django templates. Will have a go through and either make explicit when I actually want to use includes, or take it out when the context can just be assumed.